### PR TITLE
fix: Checkout auth guard

### LINKED ISSUE #12709: [part of 4.0 feature lib work] CheckoutAuthGuard fails with implicit flow or authorization code flow
When running spartacus with authorization code or implicit flow we get `undefined` as a value from `userService.get()` and that fails in the b2b roles check.

To check:
- why we get undefined there?
- is it correct behavior?

### PR DESCRIPTION
**Pull request by znikola**
closes #12709

### CODE REVIEW

File: feature-libs/checkout/components/guards/checkout-auth.guard.ts [hunk 1]
```diff
@@ -38,43 +38,51 @@ export class CheckoutAuthGuard implements CanActivate {
...
+      filter(([, , _user, isStable]) => isStable),
+      // if the user is authenticated and we have their data, OR if the user is anonymous
+      filter(([isLoggedIn, , user]) => (!!user && isLoggedIn) || !isLoggedIn),
       map(
         ([isLoggedIn, cartUser, user]: [
           boolean,
           User,
-          User | B2BUser,
+          User | B2BUser | undefined,
           boolean
```
Comment (marlass) [inline feature-libs/checkout/components/guards/checkout-auth.guard.ts]: Do we need these types?

File: feature-libs/checkout/components/guards/checkout-auth.guard.ts [hunk 2]
```diff
@@ -9,9 +9,9 @@ import {
   GlobalMessageService,
   GlobalMessageType,
   SemanticPathService,
-  User,
-  UserService,
+  User as CoreUser,
```
Comment (marlass) [inline feature-libs/checkout/components/guards/checkout-auth.guard.ts]: Can we use only model from @spartacus/core for now? Do we need 2 different interfaces?

File: feature-libs/checkout/components/guards/checkout-auth.guard.ts (hunk 1 above)
Comment (znikola) [reply feature-libs/checkout/components/guards/checkout-auth.guard.ts]: not really, removed

File: feature-libs/checkout/components/guards/checkout-auth.guard.ts (hunk 2 above)
Comment (znikola) [reply feature-libs/checkout/components/guards/checkout-auth.guard.ts]: yep, switched to the user from the user lib

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**